### PR TITLE
give the user an option to choose whether to use the intersection or union of datapoint when calculating metrics with quality standard

### PR DESCRIPTION
**Pull request by grant-Kolena**
### What change does this PR introduce and why?
This is to address Mehrad's comment [here](https://linear.app/kolena/issue/KOL-6996/[kodiak]-[voxel]-ability-to-see-qs-metrics-for-the-portion-of-the#comment-5e5a74df), we want the user to select whether to use the intersection or union of datapoint when calculating metrics with quality standard.

### Please check if the PR fulfills these requirements

- [ ] Include reference to internal ticket and/or GitHub issue "Fixes #NNNN" (if applicable)
- [ ] Relevant tests for the changes have been added
- [ ] Relevant docs have been added / updated

### CODE REVIEW

File: kolena/_experimental/quality_standard.py [hunk 1]
```diff
@@ -48,27 +48,46 @@ def _format_quality_standard_result_df(quality_standard_result: dict) -> pd.Data
...
-    )
-    krequests.raise_for_status(response)
-    return _format_quality_standard_result_df(response.json())
+    if intersect_results:
+        return _download_quality_standard_result(dataset, models, metric_groups, intersect_results)
+    else:
+        result_dfs = []
+        for model in models:
+            result_dfs.append(_download_quality_standard_result(dataset, [model], metric_groups, intersect_results))
+        return pd.concat(result_dfs, axis=1)
```
Comment (yifanwu-kolena) [inline kolena/_experimental/quality_standard.py:93]: why not use `exclude_null_results` via api but handle at the client level?

Comment (grant81) [reply kolena/_experimental/quality_standard.py:93]: For the qs request from client, we always want to exclude_null_results, this is just to choose whether the user want to only calculate metrics from datapoints that has result for all models. This is to have parity with the future FE change

Comment (grant-Kolena) [reply kolena/_experimental/quality_standard.py:93]: This is also me BTW

File: tests/integration/_experimental/test_quality_standard.py [hunk 2]
```diff
@@ -104,6 +111,11 @@ def test__download_quality_standard_result(
         quality_standard_df,
         download_quality_standard_result(dataset_name, [model_name], [metric_group_name]),
     )
+    assert_frame_equal(
+        download_quality_standard_result(dataset_name, [model_name, model_name2]),
+        download_quality_standard_result(dataset_name, [model_name, model_name2], intersect_results=False),
+        check_like=True,
+    )
```
Comment (yifanwu-kolena) [inline tests/integration/_experimental/test_quality_standard.py]: if we don't upload partial results for model2, i think this test is not that usefully

Comment (grant-Kolena) [reply tests/integration/_experimental/test_quality_standard.py]: ok, let me just separate the tests

File: tests/integration/_experimental/test_quality_standard.py [hunk 3]
```diff
@@ -164,3 +164,115 @@ def test__download_quality_standard_result(
...
+        dataset_name,
+        model_name,
+        model1_results,
+    )
+    model_2_results = model1_results[2:]
+    _upload_results(
+        dataset_name,
+        model_name2,
+        model_2_results,
+    )
```
Comment (yifanwu-kolena) [inline tests/integration/_experimental/test_quality_standard.py]: could we normalize the affix style between `model_name`, `model_name2`, `model1_results` and `model_2_results`?

**Review by yifanwu-kolena — APPROVED** [review]

**Pull request merged**